### PR TITLE
ffmpeg 2.7.1

### PR DESCRIPTION
**Pull request by denji**
###### From [changelog](http://git.videolan.org/?p=ffmpeg.git;a=blob;f=RELEASE_NOTES;hb=release/2.7), [added](http://git.videolan.org/?p=ffmpeg.git;a=blob;f=Changelog;hb=release/2.7) in ffmpeg 2.7, we can distinguish:
- Accelerate decoding of H. 264 using [MMAL](http://www.jvcref.com/files/PI/documentation/html/) (Multi-Media Abstraction Layer)
- Acceleration encode H. 264 streams means Intel QSV (Intel Quick Sync Video)
- Dropped support for version nvenc API, younger than 5.0
- For WebP images available WebPAnimEncoder API
- Improved compatibility with Quickdraw
- In bitstream filter for MPEG-4 now unpack B-frames, packaged in the style of DivX
- New decoders: TDSC, extensions with DTS lossless compression (XLL), DTS (via libdcadec), Canopus HQ/HQA
- New filters: FFT (Fast Fourier transform for video), showwavespic (showwaves filter option), detelecine and chorus
- Packer of mediacontainer (muxer) for live WebM streams
- Support accelerated decoding means Direct3D11
- Support decoding of VP9 with great depth of color and extended color spaces
- Support for Transport Layer Security (TLS)
- Supports auto rotation video on the basis specified in the metadata file
- The encoder and packer mediacontainer to APNG format associated with the file `apng`
- The nvenc encoder for HEVC/H. 265 using hardware accelerated encoder available in NVIDIA GPU
- Unpacking mediacontainer (demuxer) for Multipart JPEG

### CODE REVIEW

**Comment by DomT4** [comment]
Is 2.7 API/ABI friendly with 2.6.x?

**Comment by mistydemeo** [comment]
We'll need to test, but the major library versions have not been bumped so it's possible dynamic linkage is fine.

**Comment by denji** [comment]
@DomT4 @mistydemeo I have updated all the related packages, hope `brew-bot` finish processing for a given period of time.

**Comment by DomT4** [comment]
Let's not blindly revision everything just yet.

Can you pluck out 5 or 6 random dependants and test them locally, see if anything fails? The bot will have done a decently wide test on first commit here, which passed, so I'd rather not mass revision unless we know we have to.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by denji** [comment]
@BrewTestBot test this please

**Comment by bfontaine** [comment]
@denji It doesn’t work if you’re not a maintainer :wink:

**Comment by denji** [comment]
**mediatomb** is broken on yosemite 1

**Comment by denji** [comment]
@BrewTestBot restart only для `:mountain_lion`, timeout :package:

**Comment by denji** [comment]
@BrewTestBot restart only for `:mountain_lion`, timeout :package:

**Comment by DomT4** [comment]
I re-queued this a couple hours back and it [passed the bot](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/28304/) but for some reason the PR here hasn't updated to the green tick. Is that a problem @mikemcquaid? Do I trust the PR or the Bot in this situation? :wink:

**Comment by DomT4** [comment]
Oh, #40909 hasn't gone green either, despite the bot saying it passed, which was the build right after `ffmpeg`. Hmm.

**Comment by denji** [comment]
@DomT4 `:mountain_lion`, timeout!

**Comment by DomT4** [comment]
@denji See the new build, not the old one. It's the hyperlink in `passed the bot` in my comment.

**Comment by denji** [comment]
I am confused, perhaps we should close this PR?

**Comment by DomT4** [comment]
Nope. Let's give the test-bot some time to resolve wrapping itself in circles. There's nothing wrong with the PR itself - And the build did recently pass the bot, even if the PR hasn't realised that yet.

**Comment by mikemcquaid** [comment]
This will need rebased on `master`.

File: Library/Formula/moc.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class Moc < Formula
   desc "Terminal-based music player"
-  homepage 'http://moc.daper.net'
+  homepage "http://moc.daper.net"
   url "http://ftp.daper.net/pub/soft/moc/stable/moc-2.5.0.tar.bz2"
-  sha1 "a02c10075541995771dbdccb7f2d0ecd19d70b81"
+  sha256 "d29ea52240af76c4aa56fa293553da9d66675823e689249cee5f8a60657a6091"
+  revision 2
```
Comment (DomT4) [inline Library/Formula/moc.rb]: The `revision` is wrong here - It should be `revision 1`.

File: Library/Formula/mediatomb.rb [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
   bottle do
     sha1 "716145891055b842417844efb1f786344ba79f56" => :mavericks
     sha1 "d5fd8a1389771a2bf4677dc5f008b0c4249344aa" => :mountain_lion
     sha1 "ae3e70647489adec58c1433652d05bad3844f16a" => :lion
   end
 
-  revision 1
+  # error: explicit specialization of 'mutex' after instantiation
+  # SINGLETON_MUTEX(Timer, true);
+  depends_on MaximumMacOSRequirement => :mountain_lion
```
Comment (DomT4) [inline Library/Formula/mediatomb.rb]: Can you produce some logs for this? It build on Mavericks as recently as last November.

**Comment by mikemcquaid** [comment]
Did we ever establish if all these `revision`s were needed?

**Comment by denji** [comment]
@mikemcquaid revisions are returned to the current state.

**Comment by mikemcquaid** [comment]
@denji Thanks!

**Comment by DomT4** [comment]
The `mediatomb` breakage needs kicking upstream, but I've no objections to limiting it to Mavericks until we get a fix on that. There's little reason to block an `ffmpeg` update for it.

**Comment by mikemcquaid** [comment]
Agreed.

**Comment by mikemcquaid** [comment]
Can we strip this down just to the `ffmpeg` bump and remove the other style changes? Thanks.

**Comment by dunn** [comment]
Looks like 2.7.1 is out.

**Comment by DomT4** [comment]
We still need the other changes stripped from the PR here, `ffmpeg` bump aside. It's nice to fix style changes and such, but lumping them in with a formula that does require revisions from time to time on bumps may confuse users.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/c8a4b75b8ccec0177f745c7f51d6d0de8c8c2839. Thank you @denji!

**Pull request closed without merging**